### PR TITLE
deserialize request body to type class when body is an array/list

### PR DESCRIPTION
**Pull request by ruibritopt**

### CODE REVIEW

File: src/main/kotlin/com/github/mduesterhoeft/router/RequestHandler.kt [hunk 1]
```diff
@@ -58,6 +59,11 @@ abstract class RequestHandler : RequestHandler<APIGatewayProxyRequestEvent, APIG
         return when (requestType) {
             Unit::class -> Unit
             String::class -> input.body!!
+            List::class -> {
```
Comment (mduesterhoeft) [inline src/main/kotlin/com/github/mduesterhoeft/router/RequestHandler.kt]: Should we think a little more generic here - what about `Collection`

Comment (ruibritopt) [reply src/main/kotlin/com/github/mduesterhoeft/router/RequestHandler.kt]: If we just change `List::class` to `Collection::class` it won't work.

Comment (ruibritopt) [reply src/main/kotlin/com/github/mduesterhoeft/router/RequestHandler.kt]: https://kotlinlang.org/docs/reference/typecasts.html

Comment (mduesterhoeft) [reply src/main/kotlin/com/github/mduesterhoeft/router/RequestHandler.kt]: Oh OK - we are checking equality of the class object.

Then this should work:
```kotlin
when  {
            requestType == Unit::class -> Unit
            requestType == String::class -> input.body!!
            requestType.isSubclassOf(Collection::class) -> {
                val kClass = handler.reflect()!!.parameters.first().type.arguments.first().type!!.arguments.first().type!!.classifier as KClass<*>
                val type = TypeFactory.defaultInstance().constructParametricType(requestType.javaObjectType, kClass.javaObjectType)
                objectMapper.readValue(input.body, type)
            }
            else -> objectMapper.readValue(input.body, requestType.java)
        }
```

Comment (ruibritopt) [reply src/main/kotlin/com/github/mduesterhoeft/router/RequestHandler.kt]: That is a good approach, I was trying to do it with `is` (cast) but your way works. I updated it

**Pull request merged**